plugins/processors/dedup/dedup_test.go

```diff
@@ -0,0 +1,157 @@
+package dedup
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/stretchr/testify/assert"
+)
+
+func createMetric(name string, value int64) telegraf.Metric {
+	m, _ := metric.New(name,
+		map[string]string{"tag": "tag_value"},
+		map[string]interface{}{"value": value},
+		time.Now(),
+	)
+	return m
+}
+
+func createProcessor() Dedup {
+	return Dedup{
+		DedupInterval: internal.Duration{Duration: 2 * time.Second},
+		FlushTime:     time.Now(),
+		Cache:         make(map[uint64]telegraf.Metric),
+	}
+}
+
+func assertCacheRefresh(t *testing.T, proc *Dedup, item telegraf.Metric) {
+	id := item.HashID()
+	name := item.Name()
+	// cache is not empty
+	assert.NotEqual(t, 0, len(proc.Cache))
```

Comment:
Use `require` instead of `assert` everywhere you can, the difference is that require will halt the test and with assert it will keep going:
```
require.NotEqual(t, 0, len(proc.Cache))
```